src/components/CallView/CallView.vue

```diff
@@ -397,18 +397,6 @@ export default {
 			}
 		},
 
-		showSelectedVideo(newVal) {
-			if (newVal) {
-				this.$store.dispatch('setCallViewMode', { isGrid: false })
-			}
-		},
-
-		showSelectedScreen(newVal) {
```

Comment:
Maybe unrelated to current changes:
it is handled by code somewhere else, but it will still break, if you change any parameter with this dispatch;
https://github.com/nextcloud/spreed/blob/1ccbf67da231c37801c1406b10000144125aec35/src/store/callViewStore.js#L159-L162

For example:
`<Grid>` - start screenshare - `<Speaker>` - end screenshare - `<Grid>`
`<Grid>` - start screenshare - `<Speaker>` - open stripe `<Speaker>` - end screenshare - `<Speaker>` :warning: 

IMO, `lastIsGrid` and `lastIsStripeOpen` should only be changed, if `isGrid` or `isStripeOpen` were passed with a payload, and are different from the previous state

Reply:
Agree, and for your example, the main reason is that `clearLast` was passed as default truthy when opening the stripe.

Reply:
There is a test `callViewStore › call view mode and presentation › switching modes during presentation does not resets it after it ends`

Is it meant not to reset it ?

Reply:
Good catch, same could be added for:
- `stopPresentation` (will be overwritten anyway)
- `changeView` (same as handleClickStripeCollapse)

> Is it meant not to reset it ?

Looks like test is not checking anything. Plus isGrid and isStripeOpen can't be both true at the same time. Should be a series of test with different initial values, like `start presentation switches off grid view and restores when it ends`

Reply:
> stopPresentation (will be overwritten anyway)

Here, we are using the stored state as current state so it makes sense to empty the stored one ?

> changeView (same as handleClickStripeCollapse)

If user is updating their preferences, there is no need to keep the previous state (like resetting preferences). 

> Plus isGrid and isStripeOpen can't be both true at the same time.

I also noticed this abnormal logic. 
What cannot be the same value are isStripe and isGrid but isStripeOpen is independent from isGrid (which shouldn't be) and can have the same value.

Reply:
> Looks like test is not checking anything

regardless of the implementation, the title of the test implies that it is expected not to restore it. If we agree on restoring after closing presentation, I will change that test as we are changing the behavior.

Reply:
> If user is updating their preferences, there is no need to keep the previous state (like resetting preferences).

We also don't have to modify it twice in one call, `true` -> `null` -> `false`, that's what I'm trying to avoid

> > stopPresentation (will be overwritten anyway)

That's actually the only place where we're using lastIsGrid and lastIsStripeOpen values :warning:

Reply:
Okay, let's keep this PR for fixing issues. Refactoring (removing unnecessary updates) will come next.